beacon-chain/state/state-native/readonly_validator.go

```diff
@@ -88,6 +89,27 @@ func (v readOnlyValidator) IsNil() bool {
 	return v.validator == nil
 }
 
+// HasETH1WithdrawalCredentials returns true if the validator has an ETH1 withdrawal credentials.
+func (v readOnlyValidator) HasETH1WithdrawalCredentials() bool {
+	if v.IsNil() {
+		return false
+	}
+	return v.validator.WithdrawalCredentials[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
```

Comment:
This will panic if `WithdrawalCredentials` is nil, this could be changed in `v.IsNil` but anyway if we reach code with a nil validator entry we are already booked and there's no reason to continue running